Library/Homebrew/cask/lib/hbc/utils/quarantine.applescript

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/osascript --
+
+use framework "Foundation"
+
+on run(argv)
+  if argv's length < 3 then
+    error "Insufficient parameters: this script needs" ¬
+      & " file_path, data_url, and origin_url"
+  end if
+
+  set filepath to argv's item 1
+  set agent to "Homebrew-Cask"
+  set dataUrl to argv's item 2
+  set originUrl to argv's item 3
+
+  set quarantineProperties to { ¬
+    kLSQuarantineAgentNameKey: agent, ¬
+    kLSQuarantineTypeKey: my kLSQuarantineTypeWebDownload, ¬
+    kLSQuarantineDataURLKey: dataUrl, ¬
+    kLSQuarantineOriginURLKey: originUrl ¬
+  }
+
+  set dataLocationUrl to my NSURL's fileURLWithPath: filepath
+
+  if dataLocationUrl's checkResourceIsReachableAndReturnError: ¬
+    missing value then
+
+    if not dataLocationUrl's setResourceValue: quarantineProperties ¬
+      forKey: my NSURLQuarantinePropertiesKey ¬
+      |error|: missing value then
```

Comment:
Isn't it possible in AppleScript to get the `NSError` if this throws?

Reply:
I thought it wasn’t possible but now that you’ve mentioned it, I found out [there’s a convoluted, poorly documented way](https://www.macosxautomation.com/applescript/apps/BridgePlus_Manual/Pages/runSpotlightQuery_inFolders_error_.html) to do it!

Updated.